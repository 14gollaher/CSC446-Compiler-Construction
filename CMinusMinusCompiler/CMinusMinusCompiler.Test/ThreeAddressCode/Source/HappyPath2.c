int a, b, c;
const x = 4;
int main()
{
	int c, e;
	const z = 2.5;
	c = a + 2.2;
	e = a + b + 2.2 + z;
	c = !c;
	e = 2.25 - (z + e) * b + (4);
	a = (2 * a) + (b + c) + 2 + 3; 

	return z - (a + e) * 2;
} 