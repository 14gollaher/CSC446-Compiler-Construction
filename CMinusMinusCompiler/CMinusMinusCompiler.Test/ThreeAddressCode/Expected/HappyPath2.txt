PROC main
	_BP-8 = 2.2
	_BP-6 = _a + _BP-8
	_BP-2 = _BP-6
	_BP-12 = _a + _b
	_BP-16 = 2.2
	_BP-14 = _BP-12 + _BP-16
	_BP-22 = 2.5
	_BP-20 = _BP-14 + _BP-22
	_BP-4 = _BP-20
	_BP-2 = !_BP-2
	_BP-26 = 2.25
	_BP-32 = 2.5
	_BP-36 = _BP-32 + _BP-4
	_BP-38 = _BP-36 * _b
	_BP-30 = _BP-26 - _BP-38
	_BP-42 = 4
	_BP-40 = _BP-30 + _BP-42
	_BP-4 = _BP-40
	_BP-44 = 2
	_BP-46 = _BP-44 * _a
	_BP-50 = _b + _BP-2
	_BP-48 = _BP-46 + _BP-50
	_BP-54 = 2
	_BP-52 = _BP-48 + _BP-54
	_BP-58 = 3
	_BP-56 = _BP-52 + _BP-58
	_a = _BP-56
	_BP-60 = 2.5
	_BP-66 = _a + _BP-4
	_BP-70 = 2
	_BP-68 = _BP-66 * _BP-70
	_BP-64 = _BP-60 - _BP-68
	_AX = _BP-64
ENDP main

START PROC main